File: Arrays/kth_smallest_element.py
"""
Given an array arr[] and a number K where K is smaller than size of array, the task is to find the Kth smallest element in the given array.
note that all array elements are distinct.
"""

def kth_smallest_element(arr, k):
    if k > 0:
        arr.sort()
        print(arr[k-1])
    else:
        print(-1)
    

def test_cases():
    # expected case
    kth_smallest_element([1,4,3,2], 3) #3
    # expected case
    kth_smallest_element([6,3,9,5], 2) #5
    # k = 0
    kth_smallest_element([6,3,9,5], 0) #-1

test_cases()

File: Arrays/KthSmallestElement.java
/*
 * Given an array arr[] and a number K where K is smaller than size of array, the task is to find the Kth smallest element in the given array.
note that all array elements are distinct.
 */
import java.util.Arrays;
public class KthSmallestElement {

    public static int findSmallest(int[] arr, int k){
        
        if(k > 0){
            // sort the array
            Arrays.sort(arr);
            // return the element at the k-1 index
            return arr[k-1];
        } else {
            return -1;
        }
        
    }
    public static void main(String[] args) {
        // working test case
        System.out.println(findSmallest(new int[]{1,4,3,2}, 3)); // 3
        // working case
        System.out.println(findSmallest(new int[]{6,3,9,5}, 2)); // 5
        // k = 0
        System.out.println(findSmallest(new int[]{6,3,9,5}, 0)); // -1
    }
}
